dbt/adapters/impala/cloudera_tracking.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2022 Cloudera Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import requests
+import threading
+
+from decouple import config
+
+from dbt.events import AdapterLogger
+
+# global logger
+logger = AdapterLogger("Tracker")
+
+# global switch to trun on or off the usage tracking
+usage_tracking: bool = True
+
+
+def track_usage(tracking_payload):
+    """
+    usage tracking code - Cloudera specific
+    @param tracking_payload - list of key value pair of tracking data. Example:
+            payload = {}
+            payload["id"] = "dbt_impala_open"
+            payload["unique_hash"] = hashlib.md5(credentials.host.encode()).hexdigest()
+            payload["auth"] = auth_type
+            payload["connection_state"] = connection.state
+    """
+
+    global usage_tracking
+
+    logger.debug(f"Usage tracking flag {usage_tracking}")
+
+    # if usage_tracking is disabled, quit
+    if not usage_tracking:
```

Comment:
if this mean we have to specify usage_tracking:false in profiles.yml. we should test that path. 
Nitpick for comment: If usage tracking is disabled in profiles file , bypass instrumentation and continue with the dbt event.

Reply:
Yes, tested. To test, add usage_tracking: false in profiles.yml and the log output should have the following:

[[0m10:55:37.088996 [debug] [MainThread]: Tracker adapter: Usage tracking flag False

In case the usage_tracking: true or not provided in the profiles.yml, the log output should have the following:

[[0m10:54:22.947719 [debug] [MainThread]: Tracker adapter: Usage tracking flag True